cmd/scollector/collectors/awsBilling.go

```diff
@@ -0,0 +1,445 @@
+	getBucketObjects := &s3.ListObjectsInput{
+		Bucket: aws.String(awsBilling.bucketName),
+	}
+	//Get the objects from the bucket
+	bucketObjects, err := awsBilling.s3svc.ListObjects(getBucketObjects)
+	if err != nil {
+		return nil, err
+	}
+
+	//Sort the files found by last modified, newest first.
+	sort.Sort(sort.Reverse(ByLastModified(bucketObjects.Contents)))
+
+	//Go through the contents of the bucket and parse the different kinds of report files
+	//that that we're going to have. At the end of it we'll have a bucketContents that contains
+	//the origina S3 file, along with hopefully enough other detail.
+	var thisBucketContents []bucketContents
+	for _, bucketObject := range bucketObjects.Contents {
+		thisReport := &billingKeyStructure{}
+		thisReport.parseFromObjectKey(bucketObject.Key, awsBilling)
+		thisBucketContents = append(thisBucketContents, bucketContents{
+			origS3Item:     bucketObject,
+			awsBillingItem: thisReport,
+		})
+		//Mark files past the purge date to be purged
+		if bucketObject.LastModified.Before(awsBilling.purgeOlderThan) {
+			purgeObjects = append(purgeObjects, bucketObject)
+		}
+	}
+
+	//So, billing comes with a bunch of different manifets and whatnot. We don't really need all of that
+	//for this basic billing integration, as the first line of the CSVs we're interested in also has
+	//the breakdown of how it's all structured. We only want to process the last file that is present
+	//for each month, as each month we can get 30+ files. The most recent file for any given month should
+	//contain everything we need.
+	var monthsProcessed []string //The months that have been processed so far
+	var allBills []*billBill
+	for _, billingObject := range thisBucketContents {
+		//If the filename ends in a .gz then it's probably what we want
+		if filepath.Ext(billingObject.awsBillingItem.fileName) == ".gz" {
+			//if it ends in .gz and has the report name in it, then it's definately what we want
+			if strings.Contains(billingObject.awsBillingItem.fileName, billingObject.awsBillingItem.reportName) {
+				if !stringInSlice(billingObject.origS3Item.LastModified.Format("200601"), monthsProcessed) { //Check if we have processed this month's file already
+					//Download the file and un-gzip it in one step
+					billingFile, err := billingObject.downloadGzippedItem(awsBilling.downloader, awsBilling)
+					if err != nil {
+						return nil, err
+					}
+					allBills = append(allBills, awsBilling.ReadBillingFile(billingFile))
+					monthsProcessed = append(monthsProcessed, billingObject.origS3Item.LastModified.Format("200601")) //Log that we've processed this month
+				}
+			}
+		}
+	}
+	//Dump into scollector format and clean up
+	for _, thisBill := range allBills {
+		thisBill.toTSDB(&md)
+	}
+	//Clean up after ourselves
+	for _, purge := range purgeObjects {
+		slog.Infoln("Need to purge object", *purge.Key, "last modified", purge.LastModified)
```

Comment:
I think this info will fill the logs, seems more like a debug like so I woudl remove